### PR TITLE
Update recipe to `3.2.2`

### PR DESCRIPTION
**Pull request by ajschmidt8**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This is a continuation of #120, but fixes some CI errors.

### CODE REVIEW

**Comment by ajschmidt8** [comment]
This is a continuation of #120.

It looks like some of the patches needed to be updated and also some updates to `libffi` based on the `3.2` changes outlined in this post (see the `No longer bundle 3rd party sources` section): https://www.ruby-lang.org/en/news/2022/12/25/ruby-3-2-0-released/

@jimmynguyc, @scopatz, @sodre, @wolfv, @xhochy, it looks like builds are passing now except on `osx_64` and `win_64`.

Is anyone able to help get this over the finish line? Those failures are a bit foreign to me.

**Comment by ajschmidt8** [comment]
I'm down to just a single failing CI run now :crossed_fingers:

File: recipe/0001-Dont-t-add-SDKROOT-to-vals.patch
Comment (ajschmidt8) [inline recipe/0001-Dont-t-add-SDKROOT-to-vals.patch:1]: the changes that this patch makes have not changed. I just updated it for the `3.2.2` source code.

File: recipe/disable-backtrace-with-lines.patch
Comment (ajschmidt8) [inline recipe/disable-backtrace-with-lines.patch:1]: the changes that this patch makes have not changed. I just updated it for the `3.2.2` source code.

File: recipe/bld.bat [hunk 1]
```diff
@@ -1,7 +1,5 @@
 setlocal enableextensions
 
-RMDIR /s /q ext\fiddle\libffi-3.2.1
```
Comment (ajschmidt8) [inline recipe/bld.bat:3]: As mentioned in the release blog post below, `libffi` is no longer included in `3.2.x`. Therefore this line can be removed.

- https://www.ruby-lang.org/en/news/2022/12/25/ruby-3-2-0-released/

File: recipe/build.sh [hunk 2]
```diff
@@ -1,6 +1,5 @@
 #!/bin/bash
 # Get an updated config.sub and config.guess
-cp $BUILD_PREFIX/share/gnuconfig/config.* ./ext/fiddle/libffi-3.2.1
```
Comment (ajschmidt8) [inline recipe/build.sh:3]: As mentioned in the release blog post below, `libffi` is no longer included in `3.2.x`. Therefore this line can be removed.

- https://www.ruby-lang.org/en/news/2022/12/25/ruby-3-2-0-released/

File: recipe/build.sh [hunk 3]
```diff
@@ -16,9 +15,6 @@ export AR=$(basename $AR)
 export RANLIB=$(basename $RANLIB)
 export LD=$(basename $LD)
 
-# Remove vendored libffi
-rm -rf ext/fiddle/libffi-3.2.1
```
Comment (ajschmidt8) [inline recipe/build.sh:20]: As mentioned in the release blog post below, `libffi` is no longer included in `3.2.x`. Therefore this line can be removed.

- https://www.ruby-lang.org/en/news/2022/12/25/ruby-3-2-0-released/

File: recipe/clockid_t.patch
Comment (ajschmidt8) [inline recipe/clockid_t.patch:1]: This should fix the OS X AMD64 builds.

This patch was taken from the resources below:

- https://github.com/tsduck/tsduck/issues/630#issuecomment-683626445
- https://github.com/RIOT-OS/RIOT/pull/2811

**Comment by ajschmidt8** [comment]
success! would appreciate a review and merge whenever possible.

**Comment by wolfv** [comment]
Thanks!

**Pull request merged**